taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -102,10 +120,26 @@ const ListTranslations = ({ nodeObject, setNodeObject }) => {
                 }
             }
         })
+
+        // add shownTranslationLanguages as a tag to otherLangTags
+        shownTranslationLanguages.forEach((languageCode) => {
+            if (!addedLanguageCodes.includes(languageCode)) {
+                otherLangTags.push({'languageCode' : languageCode, 'tags' : []})
+                addedLanguageCodes.push(languageCode)
+            }
+        })
+
         // Set states
+        setAddedLanguageCodes(addedLanguageCodes);
         setMainLangRenderedTranslations(mainLangTags);
         setRenderedTranslations(otherLangTags);
-    }, [nodeObject]);
+    }, [nodeObject, shownTranslationLanguages]);
+
+    useEffect(() => {
+        // get shown translation languages from local storage
+        const shownTranslationLanguages = JSON.parse(localStorage.getItem('shownTranslationLanguages')) || [];
```

Comment:
please be aware that `JSON.parse` can throw an error. (you can test it in the console using: `JSON.parse('}')`). Also, as the local storage is exposed people can mess with it